cheetah/accelerator/dipole.py

```diff
@@ -424,6 +439,19 @@ def transfer_map(self, energy: torch.Tensor, species: Species) -> torch.Tensor:
 
         return R
 
+    def second_order_map(self, energy: torch.Tensor, species: Species) -> torch.Tensor:
+        T = base_ttensor(
+            length=self.length,
+            k1=torch.zeros_like(self.length),
+            k2=torch.zeros_like(self.length),
```

Comment:
```suggestion
            k1=torch.tensor(0.0, dtype=self.length.dtype, device=self.length.device),
            k2=torch.tensor(0.0, dtype=self.length.dtype, device=self.length.device),
```
We only need the device and dtype from `length`. If we do `zeros_like`, we would also take over the size, which might waste a lot of memory if `length` is vectorised.